### PR TITLE
Features

### PR DESCRIPTION
**Pull request by ricardo123321**
this are replicas of the Enumerable methods that can be found on ruby and were created making use of the call and yield method, I also make use of the self and have an example of the injection code, this project was a great source of information and learning about the use of the different methods you can find in ruby

### CODE REVIEW

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
+module Enumerable
+  def my_each
+    i = 0
+    arr = self
+    while i < arr.count
+      yield(arr[i])
+      i += 1
+    end
+  end
```
Comment (shshamim63) [inline enumerables.rb]: `my_each` identical and returns the same thing as ruby's [`each`](https://ruby-doc.org/core-2.6.4/Array.html#method-i-each). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)) ``
- Should return an enumerator if no block given.

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  def my_each_with_index
+    i = 0
+    arr = self
+    while index < arr.count
+      yield(arr[i], i)
+      i += 1
+    end
+  end
```
Comment (shshamim63) [inline enumerables.rb]: `my_each_with_index` identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
___
- Should return an enumerator if no block is given.
- Calls the given block once for each element in self, the block should accept two values (value, index). What is the index in your code? you have not declared it.

File: enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  def my_select
+    i = 0
+    to_change = []
+    arr = self
+    while i < arr.count
+      to_change << arr[i] if yield(arr[i])
+      i += 1
+    end
+    to_change
+  end
```
Comment (shshamim63) [inline enumerables.rb]: `my_select` identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- Should return an enumerator if no block given.

File: enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  def my_any?
+    i = 0
+    to_change = false
+    arr = self
+    while i < arr.count
+      to_change = true if yield(arr[i])
+      i += 1
+    end
+    to_change
+  end
```
Comment (shshamim63) [inline enumerables.rb]: `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- when no block or argument is given returns true if at least one of the collection is not false or nil
```ruby
[1, 2, 3].my_any? #should return true
[1, 2, nil].my_any? #should return true
[false, false, nil].my_any? #should return false
```
- when a class is passed as an argument returns true if at least one of the collection is a member of such class
```ruby
[1, 2, 3].my_any?(Integer) #should return true
[1, 'demo', false].my_any?(Integer) #should return true
['demo', false, nil].my_any?(Integer) #should return false
```
- when a Regex is passed as an argument returns false if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_any?(/t/)    #should return false.
%w[dog door rod blade].my_any?(/d/)   #should return true.
```
- when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
[1, 2, 3].my_any?(3)   #should return true
[1, 2, 2].my_any?(3)   #should return false
```

File: enumerables.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  def my_all?
+    i = 0
+    to_change = true
+    arr = self
+    while i < arr.count
+      to_change = false unless yield(arr[i])
+      i += 1
+    end
+    to_change
+  end
```
Comment (shshamim63) [inline enumerables.rb]: `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)) 
- when no block or argument is given, returns true when none of the collection members are false or nil
```ruby
[1, 2, 3].my_all? #should return true
[1, 2, nil].my_all? #should return false
[1, false, nil].my_all? #should return false
```
- when a class is passed as an argument returns true if all of the collection is a member of such class
```ruby
[1, 2, 3].my_all?(Integer) #should return true
[1, 2, nil].my_all?(Integer) #should return false
```
- when a Regex is passed as an argument returns true if all of the collection matches the Regex
```ruby
%w[dog door rod blade].my_all?(/d/)   #should return true.
%w[dog door rod blade].my_all?(/t/)   #should return false.
```
- when a pattern other than Regex or a Class is given, returns true if all of the collection matches the pattern
```ruby
[3, 3, 3].my_all?(3) #should return true
[3, 3, 1].my_all?(3) #should return false
```

File: enumerables.rb [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  def my_none?
+    i = 0
+    to_change = true
+    arr = self
+    while i < arr.count
+      to_change = false if yield(arr[i])
+      i += 1
+    end
+    to_change
+  end
```
Comment (shshamim63) [inline enumerables.rb]: `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- when no block or argument is given returns true only if none of the collection members is true
```ruby
[false, nil, false].my_none? #should return true
[1, 'demo', 2.2].my_none? #should return false
```
- when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
[1, 2, 3].my_none?(String)  #should return true
['1', '2', '3'].my_none?(String)  #should return false
```
- when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/z/)   #should return true
%w[dog door rod blade].my_none?(/d/)   #should return false
```
- when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
[1, 2, 4].my_none?(3) #should return true
[1, 2, 3].my_none?(3) #should return false
```

File: enumerables.rb [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    to_change = []
+    arr = self
+    arg = the_proc.nil?
+    while i < arr.count
+      to_change << (the_proc.call arr[i]) unless arg == true
+      to_change << yield(arr[i]) if arg == true
+      i += 1
+    end
+    to_change
+  end
```
Comment (shshamim63) [inline enumerables.rb]: `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- returns an Enumerator if no block is given
- returns a new array with the results of running block once for every element in enum
```ruby
block = proc { |num| num < (1 + 10) / 2 }
[1, 2 ,3].my_map(&block)   #should return [1, 2, 3]
[6, 7 ,5].my_map { |num| num > 5 }   #should also return [6, 7]
```

File: enumerables.rb [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+  def my_inject(fr_value)
+    i = 0
+    arr = self
+    while i < arr.count
+      fr_value = yield(fr_value, arr[i])
+      i += 1
+    end
+    fr_value
+  end
```
Comment (shshamim63) [inline enumerables.rb]: `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- combines all elements of enum by applying a binary operation, specified by a block
```ruby
[1, 2, 3].my_inject { |memo, num| memo + num } #should return 6
[1, 2, 3].my_inject(1) { |memo, num| memo + num } #should return 7
```
- when a symbol is specified combines each element of the collection by applying the symbol as a named method
```ruby
[1, 2, 3].my_inject(:+)  #should return 6
[1, 2, 3].my_inject(1, :+)  #should return 7
```

File: enumerables.rb [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+  def multiply_els
+    arr = self
+    arr.my_inject(1) { |result, obj| result * obj }
+  end
+  [2, 4, 5].multiply_els
```
Comment (shshamim63) [inline enumerables.rb]: `multiply_els` method. it uses `my_inject` as described on Odin ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/multiply_els.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)). Remember, it is not an enumerable method, so should be placed outside of the module.
```ruby
def multiply_els(arr)
  arr.my_inject(2, :*)
end
```

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :traffic_light: Required Changes :tractor: 
___
Hi @ricardo123321 :wave: ,
I liked the way you have kept your code really simple. But to get an approval of the project, you have to cover some edge case of those methods too. But the amazing thing is they are very easy to cover. Let's get to the info.
:warning: You have followed the wrong git flow. Check the remember to section of this [link](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/57421589/). I hope that you will follow the correct flow from your next project. Else the PR can be marked as invalid.
___
### :spades: Mandatory Changes :spades: 
___
1. Please cover the edge case of the methods that are mentioned below by following the comment.
- my_each
- my_each_with_index
- my_select
- my_all?
- my_none?
- my_any?
- my_map
- my_inject
2. I can not see the methods names my_count. Did you forgot to add it? `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
3. Please fix your multiply_els with the comment below.
4. The title of the PR is not self-descriptive. Can you change it with a more elaborate one?
5. The Readme does not loop interesting. Can you add more info like which document you followed to figure them out., or the names of the methods?
___
##### Please fix the issues mentioned above and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

File: enumerables.rb [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+  def my_each
+    i = 0
+    arr = self
+    if block_given?
+      while i < arr.count
+        yield(arr[i])
+        i += 1
+      end
+    else
+      arr
```
Comment (uryelah) [inline enumerables.rb]: - [ ] When no block is given my_each should return an Enumerator. 
Pay attention, an Array and an Enumerator are different classes, they are not the same.
Also remember that my_each could also work with a Range.
Make sure you return an Enumerator in your else, even if the seld if an Array or Range.
https://ruby-doc.org/core-2.6/Enumerator.html

File: enumerables.rb [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+  def my_each_with_index
+    i = 0
+    arr = self
+    if block_given?
+      while i < arr.count
+        yield(arr[i], i)
+        i += 1
+      end
+    else
+      arr
```
Comment (uryelah) [inline enumerables.rb]: - [ ] Same issue as my_each here. Return an Enumerator, not just the self. You could try to find a way to convert the self to an Enumerator for example.

File: enumerables.rb [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+    to_change = []
+    arr = self
+    if block_given?
+      while i < arr.count
+        to_change << arr[i] if yield(arr[i])
+        i += 1
+      end
+      to_change
+    else
+      arr
```
Comment (uryelah) [inline enumerables.rb]: - [ ] Same Enumerator return issue from above.

File: enumerables.rb [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+    end
+    to_change
+  end
+
+  def my_all?
+    i = 0
+    to_change = true
+    arr = self
+    while i < arr.count
+      to_change = false unless yield(arr[i])
```
Comment (uryelah) [inline enumerables.rb]: - [ ] my_all? should also work when no block is given.
[Check the documentation on all? here.](https://ruby-doc.org/core-2.7.1/Enumerable.html#method-i-all-3F)

Comment (uryelah) [reply enumerables.rb]: Ex: 
[1,3,'foo', false].all?
=> false

File: enumerables.rb [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+      i += 1
```
Comment (uryelah) [inline enumerables.rb]: - [ ] my_all should also accept a class, a regex, or a patter(like a number, string, etc), as an argument. Make sure to include those conditions in your code.

File: enumerables.rb [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+        to_change << arr[i] if yield(arr[i])
+        i += 1
+      end
+      to_change
+    else
+      arr
+    end
+  end
+
+  def my_any?(arg = nil)
```
Comment (uryelah) [inline enumerables.rb:53]: - [ ] my_any should accept a Class or Regex as arguments. Make sure to add conditions to test for those.

- [ ] my_any should also work without a block or argument, in that case it returns tru is any value at all is truthy.

File: enumerables.rb [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+    to_change = true
+    arr = self
+    while i < arr.count
+      to_change = false unless yield(arr[i])
+      i += 1
+    end
+    to_change
+  end
+
+  def my_none?
```
Comment (uryelah) [inline enumerables.rb]: - [ ] my_none should still work without a block or an argument. It returns true if all items are falsy in this case.
- [ ] Make sure to add conditions that take into account getting a Regex, Class, or another pattern as an argument.

File: enumerables.rb [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+    end
+    to_change
+  end
+
+  def my_count
+    arr = self
+    i = 0
+    total = 0
+    while i < arr.count
+      arr.my_each { total += 1 }
```
Comment (uryelah) [inline enumerables.rb]: - [ ] Please don't use the original #count method to make your my_count.
You can use something simple like length, size, or reuse your own my_each or my_each_with_index methods.

- [ ] my_count should also accept an argument and count the number of times it shows up in the self.

- [ ] my_count should also accept a block and in that case count the number of elements yielding a true value.

File: enumerables.rb [hunk 18]
```diff
@@ -0,0 +1,137 @@
...
+    end
+    to_change
+  end
+
+  def my_map(the_proc)
+    i = 0
+    to_change = []
+    arr = self
+    arg = the_proc.nil?
+    while i < arr.count
```
Comment (uryelah) [inline enumerables.rb]: - [ ] my_map should return an Enumerator if no block is given.

- [ ] my_map doesn't accept arguments, just blocks. Please remove the 'the_proc' argument.

File: enumerables.rb [hunk 19]
```diff
@@ -0,0 +1,137 @@
...
+    i = 0
+    total = 0
+    while i < arr.count
+      arr.my_each { total += 1 }
+      i += 1
+    end
+    total
+  end
+
+  def my_inject(start_value = 0, operator = nil)
```
Comment (uryelah) [inline enumerables.rb]: - [ ] my_inject should also accept a block. Make to add that condition.

- [ ] What happens if there's no start_value and only an operator is used as the sole operator? Make sure it works the same as [1,2,3,4,5].inject(:+) does.

**Review by uryelah — CHANGES_REQUESTED** [review]
Hello **Ricardo Vera Trallero**, I'm Sarah, your reviewer for this PR 👩‍🚀  

**First of all, good job with adding some more details to your code 🌟**

A lot of the fundamental behavior of each Enumerable methods is missing still.  [Make sure you read the specifications of the enumerable module, one by one, and compare it with your code](https://ruby-doc.org/core-2.7.1/Enumerable.html). 

Try also to re-use your own code, you could use my_each_with_index instead of while loops in most of the cases.

You can do it! After you are done with this your ruby foundation will be solid as a rock!💪 

---------     

## Required Changes 🏗 👷‍♀ 

----

**Comment by uryelah** [comment]
**Kindly fix the REQUIRED issues and send another code review request.**

As a tip, always test your code on the irb console every time you change a line of code. Make sure to also check if the behavior you get is the same as the original methods.
Use different arrays and ranges with different contents and lengths to test your code too.

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi @ricardo123321       
#### Great Job so far , but there are few more things required. 
### General
- [ ] `my_each_with_index`  
- it calls the given block once for each element in self
##### For example
````ruby
      my_each_output = ' ' 
      my_each_output_two = ' ' 
      block = proc { |y, x| my_each_output += "Hello: #{y}, in: #{x}\n" }
      block_two = proc { |y, x| my_each_output_two += "Hello: #{y}, in: #{x}\n" }
      [1,2,32,32,3,21,1,23,12].each_with_index(&block_two)
      [1,2,32,32,3,21,1,23,12].my_each_with_index(&block)
       p my_each_output ==  my_each_output_two ` this should return true
````
- [ ] `my_any?`:
- when no block or argument is given returns true if at least one of the collection is not false or nil
`p [nil, false, nil, false].my_any?` - should return `false`

- [ ] `my_count`
- counts the number of elements yielding a true value if a block is given
`block = proc { |num| num < (2 + 24) / 2 }`
`p [23,13,12,43].my_count(&block)` should return 1

- [ ] `my_map` 
- returns an Enumerator if no block is given:
  `p [1, 2, 3].my_map` returns `#<Enumerator>`
- [ ] `my_inject`:
- When a symbol is specified combines each element of the collection by applying the symbol as a named method
`p (5..10).my_inject(2, :*) - should return 302400`
- Combines all elements of enum by applying a binary operation, specified by a block:
````ruby
 p (1..5).my_inject(4) { |prod, n| prod * n } should return the same result as inject
````
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi @ricardo123321       
#### Great Job so far , The changes I specified earlier has not been implemented, not even one. Let fix all errors as specified previously. 
##### Note: For next project use Gitflow not Github .
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: enumerables.rb [hunk 20]
```diff
@@ -0,0 +1,165 @@
...
+    end
+    to_change
+  end
+end
+
+def multiply_els(value = 1)
+  arr = self
+  arr.my_inject(value, :*)
+end
+[2, 4, 5].multiply_els
```
Comment (shshamim63) [inline enumerables.rb]: Did you execute the code? Cause while I tried, faced an error. Based on the instruction of the multiply_els, what I can get is, it is not an enumerable method. It accepts an array as an argument and performs multiplication functionality through my_inject. So the code should looks like
```ruby
def multiply_els(array)
  array.my_inject(:*)
end
p multiply_els([1, 2, 3])
```

File: enumerables.rb [hunk 21]
```diff
@@ -0,0 +1,165 @@
...
+      end
+    end
+    if block_given?
+      my_each do |obj|
+        to_change = yield to_change, obj
+      end
+    end
+    to_change
+  end
+end
```
Comment (shshamim63) [inline enumerables.rb]: `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- combines all elements of enum by applying a binary operation, specified by a block
```ruby
%w[dog door rod blade].my_inject {|memo, word| memo.length > word.length ? memo : word}  #should return 'blade'
```
Check your to_check for this criteria. The answer is in there.
- when a symbol is specified combines each element of the collection by applying the symbol as a named method
```ruby
[1, 2, 3].my_inject(:+)  #should return 6
```
But while I tried to execute, I got an error. Debug the provided scenario and check the value of the to_change variable. You will get the error.

File: enumerables.rb [hunk 22]
```diff
@@ -0,0 +1,165 @@
...
+      case operator
+      when :+
+        arr.each { |obj| to_change += obj }
+      when :-
+        arr.each { |obj| to_change -= obj }
+      when :*
+        arr.each { |obj| to_change *= obj }
+      when :/
+        arr.each { |obj| to_change /= obj }
+      end
```
Comment (shshamim63) [inline enumerables.rb]: Don't you trust your own my_each method? If you are confused, I am assuring you, your my_each works totally fine. Now there is a better way to handle this symbol. You can use the send method. Check out this [example](https://stackoverflow.com/questions/3337285/what-does-send-do-in-ruby).

**Review by shshamim63 — CHANGES_REQUESTED** [review]
### Status :traffic_light: Changes Required :1234: 
___
Hi @ricardo123321 :wave: ,
Almost there. You just need to cover some simple and easy edge cases. Let's get to them.
___
### :spades: Mandatory Changes :spades:
___
1. Your multiply_els is not functional. So check my comments related to that and fix it.
2. my_inject is missing some edge cases. Check the comment on the code block.
3. You should not alter the default rubocop config file. check out this [link](https://github.com/microverseinc/linters-config/tree/master/ruby) for more explanation
image.
___
:warning: Based on the pull request it seems that you are not following the right gitflow, As the previous reviewer did not inform about it, so I am not raising any issue on it. But remember to follow the current gitflow in your next project else that could declare as invalid.
___
#### Please fix the issues mentioned above and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

File: enumerables.rb [hunk 23]
```diff
@@ -0,0 +1,173 @@
...
+    end
+    if block_given?
+      my_each do |obj|
+        to_change = yield to_change, obj
+      end
+    end
+    to_change
+  end
+end
```
Comment (Dandush03) [inline enumerables.rb]: I highly recommend you to fallow @uryelah  comments regarding this, basically, you've made 2 mistakes the first one would be to check the operator, there is a way more efficient way of doing it but essentially they are not relevant if you build this method properly

the second one but more important is that #my_inject only receives an operator, so the first argument you're asking is never submitted and it's crashing the method. 
maybe what you need is to find the starting of a Range and you could do so with the shift or pop method, I'll let you dig into it ;)

If you need any help with this issue, let me know thought slack on Monday

## Happy Coding

### Daniel Laloush

**Review by Dandush03 — COMMENTED** [review]
# Status: :red_circle: Required Change :red_circle: 

## Hey! :wave: 

### You've done great work so far! GOOD JOB :muscle: 

But sadly you didn't manage to succeed in the las 2 test of #my_inject method. Below i will get you some recommendations you should implement in order to get it through. Don't forget that #my_inject should behave in its core as #inject

## :red_circle: But before moving forward I should strongly advise you to create a develop branch and create a PR to it and only then submit this project. since I'm supposed to mark this project as invalid from the bad practice of git-flow which I'm not doing on this occasion since my colleagues didn't notice it so far.

check out this [link](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/) for more info

**Pull request closed without merging**